update/genesis/import.go

```diff
@@ -211,8 +232,57 @@ func newAccountCreator(accType Type) (state.AccountFactory, error) {
 	return nil, update.ErrUnknownType
 }
 
+func getTrieContainerID(accType Type) []byte {
+	switch accType {
+	case UserAccount:
+		return []byte(trieFactory.UserAccountTrie)
+	case ValidatorAccount:
+		return []byte(trieFactory.PeerAccountTrie)
+	}
+
+	return nil
+}
+
+func (si *stateImport) getTrie(shardID uint32, accType Type) (data.Trie, error) {
+	if shardID == si.shardID {
+		accountsDBType := getTrieContainerID(accType)
+		trieForShard := si.triesContainer.Get(accountsDBType)
+		return trieForShard, nil
+	}
+
+	trieForShard, ok := si.tries[shardID]
+	if ok {
+		return trieForShard, nil
+	}
+
+	dbConfig := storageFactory.GetDBFromConfig(si.storageConfig.DB)
+	dbConfig.FilePath = path.Join(si.storageConfig.DB.FilePath, core.GetShardIdString(shardID))
```

Comment:
we need to refactor this as it is very hard to test.

Reply:
yes. it is hard to mock. but you have to create temporary persisters to import those tries which are not for your shard. How about creating a component which returns a trie storage manager according to a config, which implements and interface, so it is easy to mock ?

Reply:
yup